### PR TITLE
Num paragraph

### PR DESCRIPTION
**Pull request by bond7o6**
Attempted to mimic the numbering for paragraphs in the .docx as close as possible. Just enclose your paragraph in the `point` environment and everything should work out.

### CODE REVIEW

**Comment by yjyao** [comment]
can you re sync and update the example PDF? thanks.

**Comment by yjyao** [comment]
wrong file? you changed pleading.tex but uploaded opinion.pdf.

**Comment by bond7o6** [comment]
Yea, magit was being a pain. Sorry about that.

File: engwales-plead.sty [hunk 1]
```diff
@@ -25,3 +26,10 @@
   \begin{flushright}\uline{Defendant}\end{flushright}
   \endgroup
 }
+
+%% Allow for you to list items with the number in the margin
```
Comment (yjyao) [inline engwales-plead.sty:30]: from the preview it doesn't look like it's putting the numbers on the margin. this comment is a little misleading.

File: engwales-plead.sty [hunk 2]
```diff
@@ -25,3 +26,10 @@
...
+\newcounter{point}
+\newenvironment{point}[1][]
+{\begin{enumerate}[leftmargin=4em,labelsep=1.5em]\refstepcounter{point}
```
Comment (yjyao) [inline engwales-plead.sty:33]: the enumitem package supports enumeration continuation. custom counter is unnecessary.

File: engwales-plead.sty [hunk 3]
```diff
@@ -25,3 +26,10 @@
...
+   \item[\thepoint.] #1
```
Comment (yjyao) [inline engwales-plead.sty:34]: move the label format to list

**Review by yjyao — COMMENTED** [review]
the point of introducing `point` env seems to be based on the assumption where one needs to jump in and out of the list often. may i ask where did this assumption come from?

File: engwales-plead.sty (hunk 2 above)
Comment (bond7o6) [reply engwales-plead.sty:33]: The resume option doesn't carry across sections, I thought a custom one would make sure the counting didn't reset.

**Comment by yjyao** [comment]
(i seem to be using the system incorrectly. how do i leave multiple comments in one go? should i click "start review" instead of "single comment"?

**Comment by bond7o6** [comment]
> (i seem to be using the system incorrectly. how do i leave multiple comments in one go? should i click "start review" instead of "single comment"?

I can't help you there, I tend not to use github for collaboration.

**Comment by bond7o6** [comment]
> the point of introducing `point` env seems to be based on the assumption where one needs to jump in and out of the list often. may i ask where did this assumption come from?

It seemed a more flexible solution that doesn't restrict there use. I didn't want to create it in a way that might restrict usage. Having one list environment also didn't seem like it'd be as easily readable while writing. I'm never going to actually use this as I'm not a lawyer, so it could very well be more convenient to have one list env.

File: engwales-plead.sty (hunk 3 above)
Comment (bond7o6) [reply engwales-plead.sty]: I'm not quite sure what you mean, but gave it a shot. You can now pass enumerate options through the `\begin{point}[stuff]`.

File: engwales-plead.sty (hunk 2 above)
Comment (yjyao) [reply engwales-plead.sty]: i think it does?

MWE:

```tex
\documentclass{article}
\usepackage{enumitem}
\def\items{\item foo bar baz \item foo \item foo bar}
\begin{document}
\begin{enumerate}\items\end{enumerate}
\section{hello}
\begin{enumerate}[resume]\items\end{enumerate}
\section{world}
\begin{enumerate}[resume]\items\end{enumerate}
\end{document}
```

result:

image

File: engwales-plead.sty (hunk 3 above)
Comment (yjyao) [reply engwales-plead.sty]: sorry for the confusion. i meant defining the new list with `\newlist` and `\setlist` provided by enumitem, and use `label=\arabic*.` to set the label format.

**Comment by yjyao** [comment]
> Having one list environment also didn't seem like it'd be as easily readable while writing.

if the entire document is one list then i think wrapping everything in a list is fine, much like wrapping the entire document in `document` env. alternatively, we can provide a command (instead of an environment) that opens a new "item" (basically top-level `\item` support).

on the other hand, if there's nothing else between one `point` and another, writing 

```tex
\begin{point}
blah blah
\end{point}
\begin{point}
another blah blah
\end{point}
```

would be cumbersome.

> I'm never going to actually use this as I'm not a lawyer

same here. @passamo9 can you share more about how the "Particulars of Claims" will be structured? specifically, we would like to learn:

1. is the entire document typically a huge list (like "1. blah 2. blah ...")?
2. are there going to be content in between the list items?
3. are there going to be different sections (like those you have in "Opinions.docx" and "Skeleton...docx")? if so, would you want the listing numbers to continue in a new section? imagine:

    ```
          PARTICULARS OF CLAIMS
    1. blah blah blah blah blah blah
        blah blah blah
    2. blah blah blah blah blah blah
        blah blah
                SECTION B
    3. blah blah
    ```

File: engwales-plead.sty (hunk 2 above)
Comment (bond7o6) [reply engwales-plead.sty]: Just messed around with it, and the resume option doesn't seem to work from within the point env. So I'll see about creating a new list as suggested [here](https://github.com/passamo9/LaTeX_EnglandWales_Templates/pull/10#discussion_r575738939), but I might wait for passamo9 to weigh in below on the use of it.

**Comment by bond7o6** [comment]
Most LaTeX editors allow for either automatically pair `\begin`s with `\end`s or macros which could cut down on typing, but I see the point. One thing to reduce typing could be to make it a command instead of an environment, so `\point{text here}` instead of the begin&end pairs. I'll see what @passamo9 prefers the look of though.

**Comment by passamo9** [comment]
> same here. @passamo9 can you share more about how the "Particulars of Claims" will be structured? specifically, we would like to learn:
> 
> 1. is the entire document typically a huge list (like "1. blah 2. blah ...")?

Yes, that's the case. In a Particulars of Claim, every paragraph needs to be numbered.

> 2. are there going to be content in between the list items?

Generally no, but when negligence is pleaded, the style is to write:

> 4. The accident was caused by the negligence of the Defendant, in that he
> \\
> **Particulars of Negligence** [this is centred]
> \\
>  (a) Failed to x
>  (b) Failed to y
>  (c) Failed in all circumstances to take reasonable care.


> 3. are there going to be different sections (like those you have in "Opinions.docx" and "Skeleton...docx")? if so, would you want the listing numbers to continue in a new section? imagine:
>    ```
>          PARTICULARS OF CLAIMS
>    1. blah blah blah blah blah blah
>        blah blah blah
>    2. blah blah blah blah blah blah
>        blah blah
>                SECTION B
>    3. blah blah
>    ```

Generally no, apart from the way that I described the form used in Negligence claims. The rule is that for all documents, the paragraph numbers should never repeat.

[This is an example of a more complex claim form, so there is no need to copy this style](https://www.carter-ruck.com/images/uploads/documents/MEE-CF_and_APoC.pdf). But what you can see is that instead of using (a), (b), etc. the Carter Ruck law firm uses 6.1, 6.2, etc.

Thank you so very much!

**Comment by yjyao** [comment]
thanks for the input. and the example document was pretty informational as well.

the negligence example was a case of sub-listing, which i would consider still inside the main list.

so to summarize:

1. pretty much everything in the Particulars of Claim will be in a non-interrupted list.
2. list continuation across sections is not needed
2. you'll need sub-lists, which we can cover later.
3. i noticed in the example doc they had items numbered "4A"/"4B." is this commonly required? we might need efforts to support this.

**Comment by passamo9** [comment]
1. Yes that is correct
2. Could you kindly explain this in layman terms? Thank you so much
3. Perfect
4. This is not commonly required, but it is the style in amended Particulars of Claim, when there are additional paragraphs to be added. I would not worry with that now.

**Comment by bond7o6** [comment]
For 2., they mean supporting a situation like:
```
* section 1
1. An item
2. another one
* section 2
3. a third item
```

**Comment by passamo9** [comment]
@bond7o6 Yes, 2. as you describe it would be ideal

That's exactly how it should be.

Just as a reference, here is the most famous opinion written in recent time: [The Three Knights Opinion](https://www.bindmans.com/uploads/files/documents/Final_Article_50_Opinion_10.2.17.pdf). (Just by way of background, this opinion was written by three barristers with knighthoods on Article 50 pre Brexit).

I will write a guidance document for a feature that in my opinion would be necessary to add for both the Skeleton and the Opinion—which is the repetition of "Issues" (or "Questions" in the 3 Kn Op) first identified in an earlier paragraph, then repeated, either (generally) italicised in Skeletons or (in the 3 Kn Op) just before paras 13, 36, and 63.

This is wonderful work! Thank you so much @bond7o6. I really appreciate the attention to detail

**Comment by bond7o6** [comment]
Ahhh, from your early comments it seemed like that feature wouldn't be needed. Thank you for clarifying, also thank you for the expanded example. A reworking of `point` will definitely be needed to easily accommodate everything in there.

**Comment by passamo9** [comment]
I'm so sorry for my misunderstanding—the speed at which you and the other contributors are running at is absolutely astounding!!
It's like being on a velodrome with training wheels while Tour de France cyclists are going around. Yes, we're on the same track..! But I can't say for certain that we're on the same lap 😂

**Comment by bond7o6** [comment]
No worries, just good we caught it. And don't worry to much, LaTeX is a bit irritating to get into at the start, but it'll get easier.

**Comment by passamo9** [comment]
I think that the more comments that are included in the code will make it easier for me to follow what each part means. The more you break it down Barney style the easier uptake these templates will have.

But I have to thank you so much for contributing! It's really something amazing.

**Comment by yjyao** [comment]
> list continuation across sections is not needed

sorry i got confused. i reread everything and i think this is needed. because "for all documents, the paragraph numbers should never repeat" and list continuation was used in The Three Knights Opinion given in [this comment](https://github.com/passamo9/LaTeX_EnglandWales_Templates/pull/10#issuecomment-778783758).

@passamo9 i guess then it becomes:

```tex
\begin{points}
\item We are asked to advise ...
\end{points}

\section{summary}
\begin{points}
\item Our advice, in summary ...
  \begin{points}
  \item ...
  \end{points}
\end{points}

\section{context}
\begin{points}
\item Parliament is considering ...
\item On \date{2017}{2}{8} the Bill ...
\end{points}
```

would you agree? i think this can be achieved with

```tex
\newlist{points}{
  enumerate  % base style
}{
  3 % levels supported
}
\setlist{points}{
  label=\arabic*.,
  resume,
}
```

**Comment by passamo9** [comment]
That is my mistake. My sincere apologies.

**Comment by bond7o6** [comment]
I've deleted the pdf to resolve conflicts and allow for building through CI

**Comment by yjyao** [comment]
why is this closed? i think we [concluded](https://github.com/passamo9/LaTeX_EnglandWales_Templates/pull/10#issuecomment-778842367) this will be useful?

**Comment by yjyao** [comment]
> why is this closed? i think we [concluded](https://github.com/passamo9/LaTeX_EnglandWales_Templates/pull/10#issuecomment-778842367) this will be useful?

ping @bond7o6

**Comment by passamo9** [comment]
Should I have reopened this?

**Comment by bond7o6** [comment]
It was closed accidentally because I deleted the branch. Though I was gonna implement this in the class PR. You can accept this, but perhaps @passamo9 can look over the last items left to do in #31, like the small caps, emails, and if everything looks good in the pdf. Then we can have this properly torn apart and criticised in a later PR.
